internal/backend/store/store.go

```diff
@@ -65,6 +80,35 @@ func (s *store) Mutate(ctx context.Context, pid sdktypes.ProjectID, key, op stri
 		}
 
 		if r.write {
+			if s.cfg.MaxValueSize != 0 && next.ProtoSize() > s.cfg.MaxValueSize {
+				return sdkerrors.NewInvalidArgumentError("value size %d exceeds maximum allowed size %d for a single value", next.ProtoSize(), s.cfg.MaxValueSize)
+			}
+
+			if maxCount := int64(s.cfg.MaxStoreKeysPerProject); maxCount > 0 {
+				isNewKey := false
+				if r.read {
+					isNewKey = curr.IsNothing()
+				} else {
+					has, err := tx.HasStoreKey(ctx, pid, key)
+					if err != nil {
+						return fmt.Errorf("has: %w", err)
+					}
+
+					isNewKey = !has
+				}
```

Comment:
All this validation code (up to line 111 - couldn't make the mouse get there) should be in a different method.

Reply:
done